libdnf/solv/id_queue.cpp

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright (C) 2021 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "id_queue.hpp"
+
+extern "C" {
+#include <solv/util.h>
+}
+
+
+namespace libdnf::solv {
+
+static int id_queue_sort_callback(const void * id_a, const void * id_b, void * data) {
+    auto & functor = *static_cast<std::function<int(const Id & a, const Id & b)> *>(data);
+    return functor(*static_cast<const Id *>(id_a), *static_cast<const Id *>(id_b));
```

Comment:
This solution adds another call. With a longer queue, the callback is called frequently. If the comparison function is simple (eg one comparison) the added overhead can be considerable. But let's leave it at that for now.

Reply:
Yeah, I know. I still opted for code simplicity in this case.

Reply:
Thank you for the change. Extra call in heavily used callback was not a good idea.